tools/login/actions/lostpassword.php

```diff
@@ -17,53 +17,53 @@
 
 $error = false;
 $step = 'emailForm'; // Formulaire par defaut
-$user = new \YesWiki\User($this);
+
 
 if (isset($_POST['subStep']) && !isset($_GET['a'])) { // Sub step
 	switch ($_POST['subStep']) {
 		case 1:
 			// we just submitted an email or username for verification
-			$error = !($user->loadByEmailFromDB($_POST['email']));
+			$error = !($this->user->loadByEmailFromDB($_POST['email']));
 			if ($error) {
 				$step = 'userNotFound';
 			} else {
 				$step = 'successPage';
-				$user->sendPasswordRecoveryEmail();
+				$this->user->sendPasswordRecoveryEmail();
 			}
 			break;
-		case 2:
+        case 2:
 			// we are submitting a new password (only for encrypted)
 			if ($_POST['userID'] == '' || $_POST['key'] == '') {
-				header('location: login.php');
+                $this->Redirect($this->href("", $this->config['root_page']));
 			}
 			if ((strcmp($_POST['pw0'], $_POST['pw1']) != 0) || (trim($_POST['pw0']) == '')) { // No pw0 or different pwd
 				$error = true;
-				$user->loadByNameFromDB($_POST['userID']);
+				$this->user->loadByNameFromDB($_POST['userID']);
 				$step = 'recoverForm';
 			} else {
-				$error = false;
+                $error = false;
 				$step = 'recoverSuccess';
-				if ($user->loadByNameFromDB($_POST['userID'])) {
-					if ($user->resetPassword($_POST['userID'], $_POST['key'], $_POST['pw0'])) { // entails password recovery key ckeck
-						$user->logIn(); // Set session cookie
+				if ($this->user->loadByNameFromDB($_POST['userID'])) {
+                    if ($this->user->resetPassword($_POST['userID'], $_POST['key'], $_POST['pw0'])) { // entails password recovery key ckeck
+						$this->user->logIn(); // Set session cookie
 					} else { // Not able to update password
-						$error = true;
+                        $error = true;
 					}
 				} else { // Not able to load the user from DB
-					$error = true;
+                    $error = true;
 				}
 			}
 			break;
 		} // End switch
 } elseif (isset($_GET['a']) && $_GET['a'] == 'recover' && $_GET['email'] != '') {
     $step = 'invalidKey';
-    $result = $user->checkEmailKey($_GET['email'], $_GET['u']);
+    $result = $this->user->checkEmailKey($_GET['email'], $_GET['u']);
     if ($result == false) {
         $error = true;
         $step = 'invalidKey';
     } else {
         $error = false;
-        $user->loadByNameFromDB(base64_decode($_GET['u']));
+        $this->user->loadByNameFromDB(base64_decode($_GET['u']));
```

Comment:
// le paramètre 'u' doit être décodé sinon checkemailKey ne fonctionne pas et provoque une erreur clé invalide
$result = $this->user->checkEmailKey($_GET['email'],base64_decode( $_GET['u']));